### PR TITLE
Example of plotting with Patternfly Charts 

### PR DESCRIPTION
**Pull request by mvollmer**
image

#### Notes

- This works and looks great, but takes significantly more CPU than flot, I think.

- This re-uses a lot of lib/plot.js to produce the actual numbers. I think we should keep doing that.

- With the current approach, we need to convert from the flot style data layout to the ChartArea style layout for each render. That can go away once we drop flot totally, of course, and let lib/plot.js produce data in the right format directly.

- Stacking graphs also works, but I haven't done a full instance plot yet.

- We need to compute all tick values explicitly (instead of letting ChartAxis choose them), but that's okay.  It's not hard and we need that level of control anyway.  The example does only minute ticks, but it should be straighforward to generalize this to our usual day, month and year ticks.

- By default, a Chart is "responsive" which means that it automatically scales with its parent DOM element. But this will scale the text labels and can't change the aspect ratio. So we need to take full control over the layout of the plot, like we did with flot.  This probably leads to some flicker while resizing, but I'd rather have some temporary ugliness during resize than permanent ugliness afterwards.

- Performance is unacceptable. :-( The Victory library doesn't seem to like large datasets, and spends unacceptable amounts of time prawling through them, sorting and converting, etc.  By large I mean 3 times 300 points in a stacked graph. We don't use any of the clever features of Victory, so all that processing is wasted for us.

- Options: d3.js, SVG, canvas, or just keep using flot.  D3 feels like a step backwards towards jQuery.  Canvas and SVG both seem reasonable; I think we have all data in a form that makes actually rendering it almost trivial. This would give us enough control to produce ultra nice graphs.

#### Todo

- [x] Resizing
- [x] Stacked instance plot
- [x] History controls
- [x] IO rate axis labels, common unit on top
- [x] Zooming in via selection
- [x] Icons for zoom controls
- [x] font-family: var(--pf-global--FontFamily--monospace) for axis labels
- [x] Nicer IO rate label, decimal multiplies of a power of 1024 unit.
- [x] Hovering
- [x] More than one plot with common controls and axis
- [x] Change metric instances without destroying the whole plot
- [x] Proper layout
- [x] Profiling
- [ ] Code untangling

### CODE REVIEW

File: pkg/lib/plot.js [hunk 1]
```diff
@@ -479,19 +481,23 @@ export class Plot {
...
+        if (this.element) {
+            $(this.element).on('plothover', null, this, this.hover_on);
+            $(this.element).on('mouseleave', null, this, this.hover_off);
+            $(this.element).on('plotselecting', null, this, this.selecting);
+            $(this.element).on('plotselected', null, this, this.selected);
 
-        // for testing
-        $(this.element).data('flot_data', this.flot_data);
+            // for testing
+            $(this.element).data('flot_data', this.flot_data);
```
Comment (garrett) [inline pkg/lib/plot.js:491]: Right now, this looks up `this.element` using `$()` each time. It's better to chain

```js
$(this.element).on('plothover', null, this, this.hover_on)
  .on('plotselecting', null, this, this.selecting)
  .on('mouseleave', null, this, this.hover_off)
  .on('plotselected', null, this, this.selected)
  .on('plotselecting', null, this, this.selecting)
  .on('plotselected', null, this, this.selected); 
```

..._or_ store the results of $(this.element) in a variable and use that.

File: pkg/lib/plot.js [hunk 2]
```diff
@@ -598,12 +604,14 @@ export class Plot {
         this.series = [];
         this.flot_data = [];
         this.flot = null;
-        $(this.element).empty();
-        $(this.element).data('flot_data', null);
+        if (this.element) {
+            $(this.element).empty();
+            $(this.element).data('flot_data', null);
```
Comment (garrett) [inline pkg/lib/plot.js:609]: Better as:

```js
            $(this.element).empty().data('flot_data', null);
```

**Comment by garrett** [comment]
Added missing CSS in 3c9efdb.

Looking at overriding the font now.

File: pkg/lib/plot.js (hunk 1 above)
Comment (mvollmer) [reply pkg/lib/plot.js:491]: This is old code that we are moving away from.  I wouldn't worry about it.

**Comment by mvollmer** [comment]
We will go with #14913. Patternfly Charts take too much CPU massaging the data.

**Pull request closed without merging**